nuc_morph_analysis/analyses/linear_regression/linear_regression.py

```diff
@@ -48,22 +77,41 @@ def fit_linear_regression(
     alpha - hyperparameter for lasso
     tol - tolerance to check drop in r^2 for finding best alpha (ex. 0.02)
     save_path - location to save files
+    save - whether to save movies and pngs
+    permute_col - list of features to permute and replace with noise
     """
     sns.set_context("talk")
     random_state = 2652124
 
     # init empty dicts and lists
     all_test_sc = []
     all_coef_alpha = []
-    all_perms = {'score': [], 'perm_score_mean': [], 'perm_score_std': [], 'p_value': [], 'alpha': []}
+    all_perms = {
+        "score": [],
+        "perm_score_mean": [],
+        "perm_score_std": [],
+        "p_value": [],
+        "alpha": [],
+    }
+
+    if multiple_predictions:
+        # remove 0 alpha due to convergence errors
+        alpha = [i for i in alpha if i != 0]
+        alpha = [round(i, 1) for i in alpha] 
 
     # find best alpha for Lasso model
-    for alpha_ind, this_alpha in enumerate(alpha):
-        print("fitting alpha", this_alpha)
-
+    for alpha_ind, this_alpha in tqdm(enumerate(alpha), total=len(alpha)):
         # drop any nan rows
         dropna_cols = cols + [target]
         data = data.dropna(subset=dropna_cols)
```

Comment:
this step (dropna) should be removed from the loop since the input data should not change

Reply:
Sure we can do that. I added to the documenation that the input features must not contain nan rows as well.